File: src/model/note.py
#!/usr/bin/env python3
# coding: utf-8

"""
Note class
"""

class Note:
    """"""
    def __init__(self, note_id: int, title: str, content: str, creation_date: int, update_date: int):
        self.id = note_id
        self.title = title
        self.content = content
        self.creation_date = creation_date
        self.update_date = update_date

    def __str__(self):
        return "{}:{}".format(self.id, self.title)

    def __repr__(self):
        return "{}:{}".format(self.id, self.title)

    def __hash__(self):
        return hash(self.id)

    def __eq__(self, other):
        return self.id == other.id

File: ios/Ononoki/src/model/Note.swift
import Foundation

final class Note: Identifiable, Hashable {
	let id: Int
	var title: String
	var content: String
	let creationDate: UInt64
	let updateDate: UInt64

	init(id: Int, title: String, content: String, creationDate: UInt64, updateDate: UInt64) {
		self.id = id
		self.title = title
		self.content = content
		self.creationDate = creationDate
		self.updateDate = updateDate
	}

	func hash(into hasher: inout Hasher) {
		hasher.combine(self.id)
	}
}

func ==(rhs: Note, lhs: Note) -> Bool {
	return rhs.id == lhs.id
}
